cli/internal/app/upload_app_source.go

```diff
@@ -0,0 +1,27 @@
+package app
+
+import (
+	"errors"
+	"io"
+	"net/http"
+)
+
+var ErrAppSourceUpload = errors.New("error uploading app source")
+
+func (s *Service) UploadAppSource(uploadURL string, archive io.Reader) error {
+	req, err := http.NewRequest(http.MethodPut, uploadURL, archive)
```

Comment:
AWS cannot understand the reader as it need content-length. Copying the content into a buffer fixes the issue.
If you dont, you get statusCode 501
`
	var buf bytes.Buffer
	if _, err := io.Copy(&buf, archive); err != nil {
		return err
	}

	req, err := http.NewRequest(http.MethodPut, uploadURL, &buf)
`

Reply:
done!